data/abilities.js

```diff
@@ -840,11 +840,17 @@ exports.BattleAbilities = {
 	},
 	"emergencyexit": {
 		shortDesc: "This Pokemon switches out when it reaches 1/2 or less of its maximum HP.",
-		// TODO
 		id: "emergencyexit",
 		name: "Emergency Exit",
 		rating: 3,
 		num: 194,
+		onHit: function (target) {
```

Comment:
I'm not sure that this is the best event to use; `onAfterDamage` might be more suitable.